pkg/providers/vsphere/network/network.go

```diff
@@ -660,22 +660,25 @@ func vpcSubnetPortToResult(
 	}
 
 	ipConfigs := []NetworkInterfaceIPConfig{}
+	if macAddr := subnetPort.Status.NetworkInterfaceConfig.MACAddress; len(macAddr) == 0 {
```

Comment:
I don't think there is a need for the temp `macAddr` var here.

We could prb make this a little clearer by something like so we don't have an empty if block.

```
// An empty MAC address is NSX ...
if len(subnetPort.Status.NetworkInterfaceConfig.MACAddress) != 0 {
  // IPAddresses have CIDR format.
...
}
```

Maybe save off `subnetPort.Status.NetworkInterfaceConfig.MACAddress` to temp `macAddr` var so we don't have repeat it when filling out the `NetworkInterfaceResult` but that's not too big of a deal.

That all said, why use the the MAC address to check for DHCP? Could IPs be not empty with DHCP? No IPs is a clearer indication that will be using DHCP.

Reply:
Only because https://github.com/vmware-tanzu/nsx-operator/issues/559. There will  change `ipAddress: /26` to nil soon. And I don't have a preference on checking macAddress or ipAddress. If you feel strongly on checking ipAddress, then I will wait for above MR to be merged in nsx-operator first.

Reply:
Let's wait for that MR and use `IpAddresses` instead